mono/metadata/domain.c

```diff
@@ -573,26 +572,27 @@ mono_init_internal (const char *filename, const char *exe_filename, const char *
 		mono_fixup_exe_image (exe_image);
 #endif
 	} else if (runtime_version != NULL) {
-		runtimes [0] = get_runtime_by_version (runtime_version);
-		runtimes [1] = NULL;
+		runtimes = g_slist_prepend (runtimes, (gpointer)get_runtime_by_version (DEFAULT_RUNTIME_VERSION));
 	}
 
-	if (runtimes [0] == NULL) {
+	if (runtimes == NULL) {
 		const MonoRuntimeInfo *default_runtime = get_runtime_by_version (DEFAULT_RUNTIME_VERSION);
-		runtimes [0] = default_runtime;
-		runtimes [1] = NULL;
+		runtimes = g_slist_prepend (runtimes, (gpointer)default_runtime);
 		g_print ("WARNING: The runtime version supported by this application is unavailable.\n");
 		g_print ("Using default runtime: %s\n", default_runtime->runtime_version); 
 	}
 
 	/* The selected runtime will be the first one for which there is a mscrolib.dll */
-	for (n = 0; runtimes [n] != NULL && ass == NULL; n++) {
-		current_runtime = runtimes [n];
+	GSList *tmp = runtimes;
+	while (tmp != NULL) {
```

Comment:
g_slist_foreach?

Reply:
I'm not sure a new function is really necessary considering how short this is, and it's not like C has anonymous functions. I can change it if you think there's a good reason to, though.

Reply:
Actually looking again, I think foreach would further complicate the logic thanks to the break, so I'm inclined to say it's definitely not the way to go.

Reply:
> I can change it if you think there's a good reason to, though.

Was just wondering. Yeah, I completely missed the break statement. Can't mark my own comment as resolved.